### PR TITLE
Add topology for cometlake

### PR DESCRIPTION
**Pull request by RanderWang**
(1)  RT700 is connected to sdw header1 on CometLake platform
    instead of head0 on IceLake platform, so these two platforms
    can't share the same topology file.

(2) fix compiling error for icl

### CODE REVIEW

**Comment by RanderWang** [comment]
add a temp patch to use 32bit pipeline on icelake for QA reported she can't hear any sound and can't go on testing.

**Comment by RanderWang** [comment]
16bit also failed. I am working on it.

File: tools/topology/sof-icl-rt700.m4 [hunk 1]
```diff
@@ -6,7 +6,7 @@
 include(`utils.m4')
 include(`dai.m4')
 include(`pipeline.m4')
-include(`alh.m4')
+#include(`alh.m4')
```
Comment (plbossart) [inline tools/topology/sof-icl-rt700.m4:9]: I forgot to add the file in git... But I can't remember if/why it's needed. I think I did this with @ranj063

````
more platform/common/alh.m4
divert(-1)

dnl ALH related macros

dnl ALH_CONFIG_DATA(type, idx)
define(`ALH_CONFIG_DATA',
`SectionVendorTuples."'N_DAI_CONFIG($1$2)`_tuples" {'
`	tokens "sof_alh_tokens"'
`	tuples."word" {'
`	}'
`	tuples."short" {'
`	}'
`}'
`SectionData."'N_DAI_CONFIG($1$2)`_data" {'
`	tuples "'N_DAI_CONFIG($1$2)`_tuples"'
`}'
)

divert(0)dnl
````

Comment (RanderWang) [reply tools/topology/sof-icl-rt700.m4:9]: According to @bardliao,  alh.m4 is not used.

File: tools/topology/sof-cml-rt700.m4 [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+include(`pipeline.m4')
+
+# Include TLV library
+include(`common/tlv.m4')
+
+# Include Token library
+include(`sof/tokens.m4')
+
+# Include ICL DSP configuration
+include(`platform/intel/icl.m4')
```
Comment (plbossart) [inline tools/topology/sof-cml-rt700.m4]: this is CML, why do we include ICL?

File: tools/topology/sof-cml-rt700.m4 [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+include(`platform/intel/dmic.m4')
+
+DEBUG_START
+
+#
+# Define the pipelines
+#
+# PCM0 <---> volume <----> ALH 0 BE link 1
+# PCM2 <------------------ DMIC01 (dmic0 capture, , BE link 2)
+# PCM3 <------------------ DMIC16k (dmic16k, BE link 3)
```
Comment (plbossart) [inline tools/topology/sof-cml-rt700.m4]: the offsets and PCM values are wrong here.

File: tools/topology/sof-cml-rt700.m4 [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+
+dnl DAI_ADD(pipeline,
+dnl     pipe id, dai type, dai_index, dai_be,
+dnl     buffer, periods, format,
+dnl     frames, deadline, priority, core)
+
+# playback DAI is ALH(SDW0 PIN2) using 2 periods
+# Buffers use s32le format, with 48 frame per 1000us on core 0 with priority 0
+DAI_ADD(sof/pipe-dai-playback.m4,
+	1, ALH, 258, SDW1-Playback,
```
Comment (plbossart) [inline tools/topology/sof-cml-rt700.m4]: maybe use 0x12?

File: tools/topology/sof-cml-rt700.m4 [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+	PIPELINE_SOURCE_1, 2, s32le,
+	48, 1000, 0, 0)
+
+# capture DAI is ALH(SDW0 PIN3) using 2 periods
+# Buffers use s32le format, with 48 frame per 1000us on core 0 with priority 0
+DAI_ADD(sof/pipe-dai-capture.m4,
+	2, ALH, 259, SDW1-Capture,
```
Comment (plbossart) [inline tools/topology/sof-cml-rt700.m4]: 259 -> 0x13?

File: tools/topology/sof-cml-rt700.m4 [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+		DMIC_WORD_LENGTH(s32le), 400, DMIC, 0,
+		PDM_CONFIG(DMIC, 0, FOUR_CH_PDM0_PDM1)))
+
+# dmic16k (ID: 3)
+DAI_CONFIG(DMIC, 1, 3, dmic16k,
+	   DMIC_CONFIG(1, 500000, 4800000, 40, 60, 16000,
+		DMIC_WORD_LENGTH(s16le), 400, DMIC, 1,
+		PDM_CONFIG(DMIC, 1, STEREO_PDM0)))
+
+DEBUG_END
```
Comment (plbossart) [inline tools/topology/sof-cml-rt700.m4:125]: please align with what was done for sof-icl-rt700, there are too many differences when the only change should be a DAI name and ALH index.

**Review by plbossart — CHANGES_REQUESTED** [review]
@RanderWang there are just too many differences with ICL, let's clean all this please and align so that we have a minimal number of differences between ICL and CML, and possibly a single file where we use M4 macros to handle the link0/1 difference.

File: tools/topology/sof-cml-rt700.m4 (hunk 2 above)
Comment (RanderWang) [reply tools/topology/sof-cml-rt700.m4]: ok, I will fix it

File: tools/topology/sof-cml-rt700.m4 (hunk 3 above)
Comment (RanderWang) [reply tools/topology/sof-cml-rt700.m4]: yes,  I will fix it.

File: tools/topology/sof-cml-rt700.m4 (hunk 4 above)
Comment (RanderWang) [reply tools/topology/sof-cml-rt700.m4]: the dai index in kernel is : (link_id << 8) | d->id, and dai component should be the same as it.
So it should be 1 << 8 | 2 = 258

File: tools/topology/sof-cml-rt700.m4 (hunk 5 above)
Comment (RanderWang) [reply tools/topology/sof-cml-rt700.m4]: check above reason

File: tools/topology/sof-cml-rt700.m4 (hunk 6 above)
Comment (RanderWang) [reply tools/topology/sof-cml-rt700.m4:125]: ok, I will fix it.

File: tools/topology/sof-icl-rt700.m4 (hunk 1 above)
Comment (bardliao) [reply tools/topology/sof-icl-rt700.m4:9]: > According to @bardliao, alh.m4 is not used.

Acked. @plbossart IIRC, we did it on Poland. But we didn't use it eventually. We did everything in dai.m4.

**Comment by plbossart** [comment]
@RanderWang I added some cleanups to dev/soundwire-3, it seems that there were still spurious differences and a mistake on PCM1 (was PCM0). Please double-check.

**Pull request merged**